src/EditorFeatures/TestUtilities/Workspaces/TestWorkspaceFixture.cs

```diff
@@ -19,6 +19,11 @@ public TestWorkspace GetWorkspace()
             return _workspace;
         }
 
+        public void SetWorkspace(TestWorkspace workspace)
```

Comment:
I think this would defeat the point of the fixture. This is to share a workspace between multiple tests. Mutating is uber-bad.

Reply:
I created an inner test class with a single test, so I think it's safe.
That said, I didn't like having to do that. The problem is that all the good test machinery relies on `WorkspaceFixture.GetWorkspace()` rather than that the `CreateWorkspace` in the test class :-(

Another approach I considered was creating a new derived class from `TestWorkspaceFixture`, but then I would have to re-implement a number of methods from `AbstractCSharpCompletionProviderTests` (since I couldn't derive from it anymore).

Not sure what a better design would be...